tests/integrations/test_django_1.py

```diff
@@ -44,6 +44,38 @@ def test_notify(self):
             'GET': {}
         })
 
+    def test_notify_post(self):
+        response = self.client.post('/notify/', {"test": "post"})
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(self.server.received), 1)
+        payload = self.server.received[0]['json_body']
+        event = payload['events'][0]
+        self.assertEqual(event['metaData']['request'], {
+            'method': 'POST',
+            'url': 'http://testserver/notify/',
+            'path': '/notify/',
+            'POST': {'test': ['post']},
+            'encoding': None,
+            'GET': {}
+        })
+
+    def test_notify_json_post(self):
+        response = self.client.post('/notify/', {"test": "post"},
```

Comment:
I think this is what's causing the need for the string replace — this parameter isn't being converted to valid JSON because it's not a string:

```suggestion
        response = self.client.post('/notify/', '{"test": "post"}',
```

Django seems to just converts the dict to a string, which isn't valid JSON because it has single quotes around the keys/values:

```
>>> str({"test": "post"})
"{'test': 'post'}"
```

The string replace will break if the JSON has a `'` in it:

```
>>> import json
>>> body = """{"key": "val'ue"}"""
>>> body
'{"key": "val\'ue"}'
>>> json.loads(body)
{'key': "val'ue"}
>>> replaced = body.replace("'", '"')
>>> replaced
'{"key": "val"ue"}'
>>> json.loads(replaced)
Traceback (most recent call last):
...
json.decoder.JSONDecodeError: Expecting ',' delimiter: line 1 column 14 (char 13)
```

Reply:
Fixed in 4b59d6cf5ee86e11ede693f10421c1f95379d044